test/relayHub/RelayHub.test.ts

```diff
@@ -0,0 +1,1199 @@
+                forwarder = forwarderInstance.address;
+                await token.mint('1000', forwarder);
+
+                sharedRelayRequestData = {
+                    request: {
+                        relayHub: relayHub,
+                        to: target,
+                        data: '',
+                        from: gaslessAccount.address,
+                        nonce: (await forwarderInstance.nonce()).toString(),
+                        value: '0',
+                        gas: gasLimit,
+                        tokenContract: token.address,
+                        tokenAmount: '1',
+                        tokenGas: '50000'
+                    },
+                    relayData: {
+                        gasPrice,
+                        relayWorker,
+                        callForwarder: forwarder,
+                        callVerifier: verifier,
+                        domainSeparator: getDomainSeparatorHash(
+                            forwarder,
+                            chainId
+                        )
+                    }
+                };
+
+                relayRequest = cloneRelayRequest(sharedRelayRequestData);
+                relayRequest.request.data = '0xdeadbeef';
+
+                const dataToSign = new TypedRequestData(
+                    chainId,
+                    forwarder,
+                    relayRequest
+                );
+                signature = getLocalEip712Signature(
+                    dataToSign,
+                    gaslessAccount.privateKey
+                );
+            });
+
+            it('Should fail a relayResquest if the manager is unstaked', async () => {
+                await relayHubInstance.stakeForAddress(relayManager, 1000, {
+                    value: ether('1'),
+                    from: relayOwner
+                });
+
+                let relayWorkersBefore = await relayHubInstance.workerCount(
+                    relayManager
+                );
+                assert.equal(
+                    relayWorkersBefore.toNumber(),
+                    0,
+                    `Initial workers must be zero but was ${relayWorkersBefore.toNumber()}`
+                );
+
+                await relayHubInstance.addRelayWorkers([relayWorker], {
+                    from: relayManager
+                });
+                await relayHubInstance.workerToManager(relayWorker);
+
+                await relayHubInstance.unlockStake(relayManager, {
+                    from: relayOwner
+                });
+
+                //Verifying stake is now unlocked
+                let stakeInfo = await relayHubInstance.getStakeInfo(
+                    relayManager
+                );
+                const isUnlocked = Number(stakeInfo.withdrawBlock) > 0;
+                assert.isTrue(isUnlocked, 'Stake is not unlocked');
+
+                //Moving blocks to be able to unstake
+                await evmMineMany(Number(stakeInfo.unstakeDelay));
```

Comment:
We could try to do this async or show a message that would tike a little bit longer.

Reply:
Do you mean we shouldn't wait for the blocks to be mined? If so I don't think we could avoid this. Before allowing the owner to retrieve the stake, they need to wait for a certain number of blocks to be mined, so here we are simulating exactly this.
Please have a look at the [stakeForAddress function](https://github.com/rsksmart/rif-relay-contracts/blob/master/contracts/RelayHub.sol#L312).

Reply:
What I was referring to, its instead of using an async function we could just execute the evmMineMany and keep with the flow. I have seen some tests that do that.

Reply:
Talked with Antonio about this change